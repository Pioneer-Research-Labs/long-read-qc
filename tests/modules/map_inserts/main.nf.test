nextflow_process {

    name "Test Process map_inserts"
    script "modules/map_inserts/main.nf"
    process "map_inserts"

    test("Should map inserts") {

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                input[0] = Channel.fromList([ tuple( [id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/inserts.fasta'),
                file("s3://pioneer-data/genomes/H_elongata/H_elongata_contigs.fna".toString()))])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
            ("$outputDir/Gateway_EmptyBarcodedLib1_20241217/mapped_insert_stats.tsv"),
            ("$outputDir/Gateway_EmptyBarcodedLib1_20241217/mapped_inserts.bam"),
            ("$outputDir/Gateway_EmptyBarcodedLib1_20241217/mapped_inserts.bam.bai"),
            ).match()
        }

    }

}
