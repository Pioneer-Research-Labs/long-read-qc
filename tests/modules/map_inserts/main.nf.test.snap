{
    "Should map inserts": {
        "content": [
            "/Users/sandie/projects/pioneerlabs/long-read-qc/.nf-test/tests/dce2422047bfc8e07cdbe1942a67a23a/results/Gateway_EmptyBarcodedLib1_20241217/mapped_insert_stats.tsv",
            "/Users/sandie/projects/pioneerlabs/long-read-qc/.nf-test/tests/dce2422047bfc8e07cdbe1942a67a23a/results/Gateway_EmptyBarcodedLib1_20241217/mapped_inserts.bam",
            "/Users/sandie/projects/pioneerlabs/long-read-qc/.nf-test/tests/dce2422047bfc8e07cdbe1942a67a23a/results/Gateway_EmptyBarcodedLib1_20241217/mapped_inserts.bam.bai"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-31T18:20:33.21713"
    }
}